Automattic/jetpack
# Recurring Payments: Remove BlockIcon Wrappers

This is arguably coming from a place of ignorance, so please forgive me if I'm missing something obvious, and consider this PR a request-for-comment.

#### Changes proposed in this Pull Request:

Recurring Payments: Remove `BlockIcon` Wrappers from `icon` props.

All of these are passed as an `icon` prop to [`Placeholder` components](https://developer.wordpress.org/block-editor/components/placeholder/), which accepts a string, or `ReactElement`. The `icon` we're using here is React element for an SVG: https://github.com/Automattic/jetpack/blob/ac55ffc4b899c5922f9fcfffac4d142bd3b713c5/extensions/blocks/recurring-payments/index.js#L15-L22

The [`BlockIcon` wrapper component](https://github.com/WordPress/gutenberg/blob/2e45a8b1b75d40933e73d19a4068e82235b3b7ca/packages/block-editor/src/components/block-icon/index.js) is arguably less public -- it's part of the `@wordpress/editor`, and I couldn't find any documentation for it. It mostly wraps the component tree that's passed to it in an `Icon` component.

(To be continued -- maybe we should wrap in `Icon` instead?)

- [ ] This can also be done for other blocks in the repo, since other blocks follow that pattern.

##### After

![image](https://user-images.githubusercontent.com/96308/63686851-d33ed900-c824-11e9-8842-dcf3a9aacdea.png)

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

Existing part.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Start with a site with a Jetpack bundled plan, such as a complete plan.
* Go to Posts > Add New, and add a Payment block
    * Check the different statuses of the block; icons should always be displayed properly.
